lesson_2/calculator.py

```diff
@@ -1,16 +1,44 @@
-# calculator program that will:
-    # Ask the user for two numbers.
-    # Ask the user for operation to perform: add, subtract, multiply or divide.
-    # Perform the calculation and display the results
+'''calculator program that will:
+    Ask the user for two numbers.
+    Ask the user for operation to perform: add, subtract, multiply or divide.
+    Perform the calculation and display the results'''
 
-print('Welcome to Calculator!')
+def prompt(message):
+    print(f"==> {message}")
 
-# no error for input/ will crash if input is wrong --> would add error if was function
-num1 = int(input('Enter a number: '))
-num2 = int(input('Enter another number: '))
+def invalid_input(str_num):
+    try:
+        int(str_num)
+    except ValueError:
+        return True
+    return False
```

Comment:
praise: this is great, first time seeing a try/except from you and I like the function name! I can deduce what it does just by reading the function name. If you want to take it a step further, you could name it `is_invalid_input` so that it flows nicer in your code:

```py
if invalid_input(foo):
```

vs.

```py
if is_invalid_input(foo):
```

The below example is more like natural english which reduces cognitive load when someone is reading it for the first time.